src/core.jl

```diff
@@ -241,7 +242,7 @@ draw!(img::AbstractArray{T,2}, p::CartesianIndex{2}, color::T = oneunit(T); in_b
 
 function draw!(img::AbstractArray{T,2}, y::Integer, x::Integer, color::T; in_bounds::Bool=false, thickness::Union{Integer, Nothing}=nothing) where T<:Colorant
     in_bounds ? img[point.y, point.x] = color : drawifinbounds!(img, y, x, color)
-    isnothing(thickness) || drawwiththickness!(img, y, x, color, in_bounds, thickness)
+    (isnothing(thickness) && thickness != 1) || drawwiththickness!(img, y, x, color, in_bounds, thickness)
```

Comment:
If I understand it right, the pipeline is:

1. draw a thickness=1 pixel
2. if thickness !=1, draw another pixel with thickness parameter.

We don't really need to do step 1 if `thickness!=1`, right?

The right expression is `isnothing(thickness) || thickness ==1 || drawwiththickness!(...)`

It becomes a little tricky to understand, so I'd recommend to rewrite it:
```julia
function draw!(img::AbstractArray{T,2}, y::Integer, x::Integer, color::T; in_bounds::Bool=false, thickness::Union{Integer, Nothing}=nothing) where T<:Colorant
    if !isnothing(thickness)
        drawwiththickness!(img, y, x, color, in_bounds, thickness) # recursively call drawifinbounds!
    else
        in_bounds ? img[point.y, point.x] = color : drawifinbounds!(img, y, x, color)
    end
end

function drawwiththickness!(img::AbstractArray{T,2}, y0::Int, x0::Int, color::T, in_bounds::Bool, thickness::Int) where {T<:Colorant}
    if thickness == 1
        draw!(img, y0, x0, color, in_bounds=in_bounds, thickness=nothing)
    else
        ...
    end
end
```